app/code/Magento/Review/Test/Unit/Observer/CatalogProductListCollectionAppendSummaryFieldsObserverTest.php

```diff
@@ -0,0 +1,138 @@
+     * @var CatalogProductListCollectionAppendSummaryFieldsObserver
+     */
+    private $observer;
+
+    /**
+     * @var Observer|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $observerMock;
+
+    /**
+     * @var Collection|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $productCollectionMock;
+
+    /**
+     * @var StoreInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $storeMock;
+
+    /**
+     * @var StoreManagerInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $storeManagerMock;
+
+    /**
+     * @var Summary|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $sumResourceMock;
+
+    /**
+     * @var SummaryFactory|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $sumResourceFactoryMock;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp() : void
+    {
+        $this->eventMock = $this->getMockBuilder(Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getCollection'])
+            ->getMock();
+
+        $this->observerMock = $this->createMock(Observer::class);
+
+        $this->productCollectionMock = $this->getMockBuilder(Collection::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->storeManagerMock = $this->getMockBuilder(StoreManagerInterface::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getStore'])
+            ->getMockForAbstractClass();
+
+        $this->storeMock = $this->getMockBuilder(StoreInterface::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getId'])
+            ->getMockForAbstractClass();
+
+        $this->sumResourceFactoryMock = $this->getMockBuilder(SummaryFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+
+        $this->sumResourceMock = $this->getMockBuilder(Summary::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['appendSummaryFieldsToCollection'])
+            ->getMock();
+
+        $this->observer = new CatalogProductListCollectionAppendSummaryFieldsObserver(
+            $this->sumResourceFactoryMock,
+            $this->storeManagerMock
+        );
+    }
+
+    /**
+     * Product listing test
+     */
+    public function testAddSummaryFieldToProductsCollection() : void
+    {
+        $this->eventMock->expects($this->once())
+            ->method('getCollection')
+            ->willReturn($this->productCollectionMock);
+
+        $this->observerMock->expects($this->once())
+            ->method('getEvent')
+            ->willReturn($this->eventMock);
+
+        $this->storeManagerMock
+            ->expects($this->once())
+            ->method('getStore')
+            ->willReturn($this->storeMock);
+
+        $this->storeMock
+            ->expects($this->once())
+            ->method('getId')
+            ->willReturn('1');
```

Comment:
Replace magic numbers with the corresponding const - for example `private const STUB_STORE_ID`

Reply:
**done**